src/memory.rs

```diff
@@ -0,0 +1,153 @@
+//! Utilities for converting Kubernetes quantities to Java heap settings.
+
+use k8s_openapi::apimachinery::pkg::api::resource::Quantity;
+
+use crate::error::{Error, OperatorResult};
+use std::str::FromStr;
+
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub enum BinaryMultiple {
+    Kibi,
+    Mebi,
+    Gibi,
+    Tebi,
+    Pebi,
+    Exbi,
+}
+
+impl BinaryMultiple {
+    pub fn to_legacy(&self) -> String {
+        match self {
+            BinaryMultiple::Kibi => "k".to_string(),
+            BinaryMultiple::Mebi => "m".to_string(),
+            BinaryMultiple::Gibi => "g".to_string(),
+            BinaryMultiple::Tebi => "t".to_string(),
+            BinaryMultiple::Pebi => "p".to_string(),
+            BinaryMultiple::Exbi => "e".to_string(),
+        }
+    }
+
+    pub fn upscale(&self) -> Self {
```

Comment:
What does this do? Why does `Kibi` transparently do nothing rather than fail?

Reply:
But saturating here doesn't make sense when we still multiply the amount by 1024 on the other end.

The more correct way then would be to have `BinaryAmount::upscale()` fail and the full unit-and-value scaler react to that by saturating and swallowing the error.